src/include/parser/expression/abstract_expression.h

```diff
@@ -125,9 +213,43 @@ class AbstractExpression {
   virtual void FromJson(const nlohmann::json &j);
 
  private:
-  ExpressionType expression_type_;                             // type of current expression
-  type::TypeId return_value_type_;                             // type of return value
-  std::vector<std::shared_ptr<AbstractExpression>> children_;  // list of children
+  /**
+   * Walks the expression trees and generate the correct expression name
+   */
+  virtual void DeduceExpressionName();
```

Comment:
Why is this one `Deduce` when the other ones are `Derive`?

Reply:
ah... I copied from Peloton.... I can standardize it. Which one makes more sense?

Reply:
I think `Derive` makes more sense here.